core/src/main/java/com/linecorp/armeria/client/RetryingRpcClient.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static com.linecorp.armeria.common.util.Functions.voidFunction;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.client.backoff.Backoff;
+import com.linecorp.armeria.common.DefaultRpcResponse;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.RpcResponse;
+
+import io.netty.channel.EventLoop;
+
+/**
+ * A {@link Client} decorator that handles failures of an invocation and retries RPC requests.
+ */
+public final class RetryingRpcClient extends RetryingClient<RpcRequest, RpcResponse> {
+    private static final RetryException RETRY_EXCEPTION = new RetryException("need to retry request");
```

Comment:
How about clearing the stack trace?

```java
Exceptions.clearTrace(new RetryException())
```

Also, the exception type itself gives us enough signal that we need to retry, so how about removing the message string?